File: leetcode_31.clj
(use 'clojure.string)

(defn solution [s]
  (let [s (lower-case 
           (replace s #"\W" ""))]
    (if (= s
           (reverse s))
      true
      false)))

(assert
 (= (solution "A man, a plan, a canal: Panama")
    true))

(assert
 (= (solution "race a car")
    false))

(assert
 (= (solution " ")
    true))

(println "Pass!")

File: leetcode_31.py
class Solution:
    # @param s, a string
    # @return a boolean
    def isPalindrome(self, s):
        tmp_s = ''
        for e in s:
            if e >= 'a' and e <= 'z':
                tmp_s += e
            elif e >= 'A' and e <= 'Z':
                tmp_s += e.lower()
            elif e >= '0' and e <= '9':
                tmp_s += e
        i = 0
        j = len(tmp_s) - 1
        result = True
        while i < j:
            if tmp_s[i] != tmp_s[j]:
                result = False
                break
            i += 1
            j -= 1
        return result

s = Solution()
print s.isPalindrome('1a2')


